nav2_costmap_2d/test/costmap_filter_info/costmap_filter_info_pub.cpp

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2020 Samsung Research Russia
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <memory>
+
+#include "rclcpp/rclcpp.hpp"
+#include "nav2_msgs/msg/costmap_filter_info.hpp"
+
+class CostmapFilterInfoPublisher : public rclcpp::Node
+{
+public:
+  CostmapFilterInfoPublisher()
```

Comment:
I'm not seeing where this is used. There's also nothing publishing the map_filter topic.

Reply:
According to HLD, there are two topics should be exist in the system: 
 - costmap filter info topic
 - map mask topic

The last one (mask) is easily could be published via Map Server running on mask PNG file for example. I did not included into a commit the configuation how to run Map Server because of its simplicity.

First topic (filter info) - is never being published before. For this new type of topic `nav2_costmap_2d/test/costmap_filter_info` was made as an attempt to make it easier to use costmap filter feature. Initially, I've considered to have semantic map server for this. However, before it will be developed, this test node makes all necessary `nav2_msgs::msg::CostmapFilterInfo` messaging job.

The node is used by all costmap filters to externally specify the linear mask->filter space conversion, mostly required for a `SpeedFilter` which is currently under development (for today I have working prototype which is planned to be committed after this review).

Reply:
OK. So your test would have 2 map servers running, this one and a "normal" one where the "normal one"'s topic is matched up with the output of this one's mask topic? OK makes sense. I thought this was trying to replace the map server entirely for the tests. You could just create the map server object inside this one to run since its just a node. Eg `auto node = std::make_shared<nav2_map_server::MapServer>();` like the `main` function does.

Reply:
This works too, just wasn't immediately clear to me without an actual test using it

Reply:
Yes, there are two `map_server` in this configuration: one is a standard, another is configured for publishing mask messages. I'll add launch-file into `nav2_costmap_2d/test` in next commit which launches both second map-server and `costmap_filter_info_pub` test executable.

Reply:
OK

Reply:
Added new `nav2_costmap_2d/test/test_launch_files/costmap_filter_info.launch.py` file for better UX.